src/liballoc/oom.rs

```diff
@@ -31,10 +35,21 @@ pub fn oom() -> ! {
     handler();
 }
 
+/// Common out-of-memory routine
+#[cfg(not(target_has_atomic = "ptr"))]
+#[cold]
+#[inline(never)]
+#[unstable(feature = "oom", reason = "not a scrutinized interface",
+           issue = "27700")]
+pub fn oom() -> ! {
```

Comment:
To help with the `#[cfg]` amount in this module, perhaps this could have two `mod imp` modules with the `#[cfg]` attributes and the implementation is reexported?